src/ipc/bridge.cc

```diff
@@ -736,14 +736,15 @@ static void initRouterTable (Router *router) {
     );
   });
 
-  /**
-   * Creates a directory at `path` with an optional mode.
+    /**
+   * Creates a directory at `path` with an optional mode and an optional recursive flag.
    * @param path
    * @param mode
+   * @param recursive
    * @see mkdir(2)
    */
   router->map("fs.mkdir", [](auto message, auto router, auto reply) {
-    auto err = validateMessageParameters(message, {"path", "mode"});
+    auto err = validateMessageParameters(message, {"path", "mode", "recursive"});  // Added "recursive" here
```

Comment:
Adding `"recursive"` here makes this a required argument. Is that needed?

Reply:
I want to make it an optional field in the options object in the JS api, but it looks like the JS side forces a value before calling the c++ bridge. I could be reading it wrong too. Happy to change anything.